### PR TITLE
Enhance template engine in BultInit plugin to be usable for generating samples

### PR DESCRIPTION
**Pull request by jjohannes**
The PR provides enhancements to the template engine, which can be used to generate full samples/guides with readme. This was used for: #14291

### CODE REVIEW

**Comment by jjohannes** [comment]
@bot-gradle test this

File: subprojects/build-init/src/test/groovy/org/gradle/buildinit/plugins/internal/BuildScriptBuilderKotlinTest.groovy [hunk 1]
```diff
@@ -481,7 +481,8 @@ Alternatively:
 """)
 
         when:
-        builder.create().generate()
+        builder.create(target).generate()
+        println(outputFile.text)
```
Comment (rieske) [inline subprojects/build-init/src/test/groovy/org/gradle/buildinit/plugins/internal/BuildScriptBuilderKotlinTest.groovy:485]: I guess this was left here by accident?

Comment (jjohannes) [reply subprojects/build-init/src/test/groovy/org/gradle/buildinit/plugins/internal/BuildScriptBuilderKotlinTest.groovy:485]: Yes. Good catch, thanks. Removed.

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle QF

**Comment by jjohannes** [comment]
@bot-gradle test QF

**Pull request merged**